db/db.go

```diff
@@ -18,50 +19,43 @@ const (
 	TxStatusNil TxStatus = iota
 	TxStatusConfirming
 	TxStatusConfirmed
-	// TxStatusDispatched
 	TxStatusSubmitted
 )
 
 type Scannable interface {
 	Scan(dest ...interface{}) error
 }
 
-// DB is an storage adapter building on top of a sql database. It is the place
-// where all txs details are stored. We can query details of a particular txs or
-// update its status.
+// DB is a storage adapter (built on top of a SQL database) that stores all
+// transaction details.
 type DB interface {
-
-	// Initialize the database. Init should be called once the db object is created.
+	// Initialise the database. Init should be called once the database object
+	// is created.
 	Init() error
 
-	// InsertTx inserts the tx into the database.
-	InsertTx(tx abi.Tx, tag abi.B32, gaas bool) error
-
-	// Tx gets the details of the tx with given txHash. It returns an `sql.ErrNoRows`
-	// if tx cannot be found.
-	Tx(hash abi.B32, transformed bool) (abi.Tx, error)
-
-	// Txs returns txs with the given tag.
-	Txs(tag abi.B32, page, pageSize uint64) (abi.Txs, error)
+	// InsertTx inserts the transaction into the database.
+	InsertTx(tx tx.Tx) error
```

Comment:
Consider combining this with `UpdateStatus`. What is the expected status of `tx` here, after insertion? What if `tx` already exists? It might be worth explicitly requiring a status here.

Reply:
There is an option of making the status `nil`-able such that whatever the current status is, is maintained.